arm-datalake-analytics/account/2016-11-01/swagger/account.json

```diff
@@ -0,0 +1,1597 @@
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Data Lake Analytics account to update."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/DataLakeAnalyticsAccountUpdateParameters"
+            },
+            "description": "Parameters supplied to the update Data Lake Analytics account operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the specified account.",
+            "schema": {
+              "$ref": "#/definitions/DataLakeAnalyticsAccount"
+            }
+          },
+          "201": {
+            "description": "Successfully initiated an update of the specified account.",
+            "schema": {
+              "$ref": "#/definitions/DataLakeAnalyticsAccount"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "StorageAccountProperties": {
+      "required": [
+        "accessKey"
+      ],
+      "properties": {
+        "accessKey": {
+          "type": "string",
+          "description": "the access key associated with this Azure Storage account that will be used to connect to it."
+        },
+        "suffix": {
+          "type": "string",
+          "description": "the optional suffix for the storage account."
+        }
+      },
+      "description": "Azure Storage account properties information."
+    },
+    "UpdateStorageAccountProperties": {
+      "properties": {
+        "accessKey": {
+          "type": "string",
+          "description": "the updated access key associated with this Azure Storage account that will be used to connect to it."
+        },
+        "suffix": {
+          "type": "string",
+          "description": "the optional suffix for the storage account."
+        }
+      },
+      "description": "Azure Storage account properties information to update."
+    },
+    "StorageAccountInfo": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the account name associated with the Azure storage account."
+        },
+        "properties": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "the properties associated with this storage account.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "description": "Azure Storage account information."
+    },
+    "CreateStorageAccountInfo": {
+      "required": [
+        "properties",
+        "name"
```

Comment:
No it doesn't when you are adding it during an account creation PUT. Then you MUST include the name of the storage account, otherwise ADLA account creation will fail.